include/rmm/mr/device/cuda_async_view_memory_resource.hpp

```diff
@@ -72,7 +72,8 @@ class cuda_async_view_memory_resource final : public device_memory_resource {
    */
   [[nodiscard]] cudaMemPool_t pool_handle() const noexcept { return cuda_pool_handle_; }
 
-  cuda_async_view_memory_resource() = default;
+  cuda_async_view_memory_resource()  = default;
+  ~cuda_async_view_memory_resource() = default;
```

Comment:
Fixed a missing rule-of-6 destructor declaration here, which raised a warning in my IDE.